eng/targetingpacks.targets

```diff
@@ -4,6 +4,7 @@
                             $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">
     <_UseLocalTargetingRuntimePack>true</_UseLocalTargetingRuntimePack>
     <EnableTargetingPackDownload>false</EnableTargetingPackDownload>
+    <GenerateErrorForMissingTargetingPacks>false</GenerateErrorForMissingTargetingPacks>
```

Comment:
@ericstj @Anipik why do we need to set this? It this something that changed recently in the SDK? Does the SDK error when using a local, live built targeting pack, the way we do? Assuming it doesn't as we would have hit this when we upgraded the SDK to 5.0.100.

Reply:
The SDK doesn't download the targeting pack because of EnableTargetingPackDownload.  It does still check that the ref-pack is available.  In 5.0, the value of the ref-pack used was always lower than the one in the SDK, so the SDK would permit roll forward to its own copy (only to be later replaced by these targets), but avoiding the error.  Now we're trying to use a 5.0 GA SDK with a 6.0 ref pack, so it cannot rollforward and emits the error.  This suppresses that error.

Reply:
> In 5.0, the value of the ref-pack used was always lower than the one in the SDK, so the SDK would permit roll forward to its own copy

What do you mean by value of the ref-pack? Can you please elaborate a bit on this by providing examples of which versions were used in 5.0?

Reply:
`KnownFrameworkReference` defines which version of the ref-pack is used for a given TFM (RuntimeFrameworkVersion can also override this).  Since the SDK doesn't support 6.0 yet, we're adding that support by adding a `KnownFrameworkReference` reference for it and pointing it at the 6.0 ref packs.  

In most of 5.0 we didn't override this, we just used the built in `KnownFrameworkReference` from the SDK which always had a version that match the ref-pack it redistributed.  Some projects did set RuntimeFrameworkVersion (which could be higher than the SDK) but those projects would just download the version they requested.  Early in 5.0 when we were using a pre-5.0 SDK we didn't have to worry about these targetingpack.targets that disable the download, because we weren't trying to replace the SDK's refpack resolution with our own, but I suspect we'll run into this same problem every time we try to use an SDK to target a new framework before we've moved to using an SDK that officially supports it.

Normally this "just works" for a project as the SDK will acquire that ref-pack via PackageDownload.  This isn't working for the library projects that use our own live-built refpack.  As I mentioned these targets disabled that acquisition, however they didn't disable the checking of presence of the refpack : https://github.com/dotnet/sdk/blob/44f381b62d466565639d51847c9127afbe7062a9/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets#L315-L346.  The SDK ends up checking the packs directory for this ref pack, doesn't find it and isn't permitted to download it, so it raises an error.  

Another version of this fix would be to disable that target completely, or make sure that target was probing our artifacts directory for the ref-pack rather than us just [overriding](https://github.com/dotnet/runtime/blob/96d7aecbf4b616002a6d0815cdbf125bc929012e/eng/targetingpacks.targets#L65-L80) its output.  It seems to me like given the current design of disabling the restore and overriding the result of ResolveTargetingPackAssets, we should also not care if ResolveTargetingPackAssets couldn't find it since we just rewrite its output to point to our built refpack.

Reply:
Thanks for the thorough explanation. I'm still confused why this regressed with the SDK update to 5.0.100 as we were already using the 6.0 TFM. I'm curious if this problem already existed before the SDK update but just didn't pop up as the targeting packs were already in the cache.

Reply:
> I'm curious if this problem already existed before the SDK update but just didn't pop up as the targeting packs were already in the cache.

I believe this is happening when we updated the SDK to 5.0.100 because that one has the 5.0.0 targeting pack, however the `MicrosoftNetCoreAppVersion` which we use to specify the targeting pack version for the `net6.0` known framework reference is pointing to 5.0.0-rc1.* 

We should probably update as part of this PR the `MicrosoftNetCoreAppVersion` to `5.0.0` and also in any other places where we reference the `Microsoft.NETCore.App` version `5.0.0-rc.1.20451.14`.

Reply:
Right, it was working because we were still using the 5.0 RC ref pack reference for net6.0, which the SDK was rolling forward to its own 5.0.0 ref pack.  😖

Reply:
> We should probably update as part of this PR the MicrosoftNetCoreAppVersion to 5.0.0 and also in any other places where we reference the Microsoft.NETCore.App version 5.0.0-rc.1.20451.14.

This PR is actually already updating that property, but to a 6.0.0-alpha version which confuses me. Why would we use a pre-built 6.0 reference anywhere instead of the live targeting pack for both the runtime and the apphost?

Reply:
Certainly the 6.0 version is better than 5.0-rc for representing 6.0.  Imagine the mode project's are in once we actually get an LKG SDK with 6.0 support.  They'll be using some LKG 6.0 ref-pack.

Presumably there are projects in the repo that are needing to represent themselves as "net6.0" (for example in a package) but aren't set up to build against the live ref-pack that is built (we know of some in installer, not sure if there are any in coreclr too).  I agree we should weed out those projects and fix them: left over work from runtime consolidation.  Not sure if this PR is the right place for it, as this change is blocking the ingestion to SDK.

Reply:
I extracted setting that property out into https://github.com/dotnet/runtime/pull/45858 as devs are currently blocked by it.